includes/admin.form.inc

```diff
@@ -127,23 +132,41 @@ function islandora_video_admin_submit($form, &$form_state) {
  * Form validation ensuring the selected ffmpeg options are available.
  */
 function islandora_video_admin_validate($form, &$form_state) {
-  // Ensure the mp4 audio codec is enabled.
-  $raw_value = $form_state['values']['islandora_video_mp4_audio_codec'];
-  if ($raw_value) {
-    if (preg_match('/[^0-9a-zA-Z_-]/', $raw_value) === 1) {
-      form_set_error('', "The value entered for MP4 audio codec contains forbidden characters.");
-      return;
+  // Ensure the mp4 audio codec is present if MP4 derivative is enabled.
+  if ($form_state['values']['islandora_video_make_mp4_locally'] == TRUE) {
+    $raw_value = $form_state['values']['islandora_video_mp4_audio_codec'];
+    if ($raw_value) {
+      if (preg_match('/[^0-9a-zA-Z_-]/', $raw_value) === 1) {
+        form_set_error('islandora_video_mp4_audio_codec', "The value entered for MP4 audio codec contains forbidden characters.");
+        return;
+      }
+      // Lower-case the value because all codecs in ffmpeg are lower case.
+      $codec = strtolower(trim($raw_value));
+      $form_state['values']['islandora_video_mp4_audio_codec'] = $codec;
+
+      // Test that FFMPEG path is valid.
+      $ffmpeg = ($form_state['values']['islandora_video_ffmpeg_path'] !== variable_get('islandora_video_ffmpeg_path', 'ffmpeg') ? $form_state['values']['islandora_video_ffmpeg_path'] : variable_get('islandora_video_ffmpeg_path', 'ffmpeg'));
+      if (preg_match('/[^0-9a-zA-Z\\/\\\\_-]/', $ffmpeg) === 1) {
+        form_set_error('islandora_video_ffmpeg_path', "The value entered for FFmpeg path contains forbidden characters.");
+        return;
+      }
+      $command = $ffmpeg . ' -version';
+      exec($command, $output, $ret);
+      if ($ret !== 0) {
+        form_set_error('islandora_video_ffmpeg_path', "FFmpeg was not found. A valid FFmpeg path is required if MP4 derivative generation is enabled.");
+      }
+      else {
+        // Test that the specified codec is present.
+        unset($output);
+        $command = $ffmpeg . ' -encoders 2>/dev/null | grep "^ ...... ' . $codec . ' "';
+        exec($command, $output, $ret);
+        if (!$output) {
+          form_set_error('islandora_video_mp4_audio_codec', 'The selected MP4 codec was not found in ffmpeg. Try using aac or enable the desired codec.');
```

Comment:
Do you mean display the input in the error message? Or to change the suggestion from aac to the default value libfaac (unfortunately aac is the one most likely available because it's natively bundled with ffmpeg, but the default for legacy reasons is libfaac.)

Reply:
Ok, so i'm confused (not your fault of course). We suggest people to use generic build in code "aac", but we default to "libfaac" which is deprecated and not supported anymore? https://trac.ffmpeg.org/wiki/Encode/AAC What a strange world. Anyway, whatever we suggest v/s what we default, I kinda feel it should be the same but maybe i'm wrong. Also funny about that is that aac "codec" and "compression standard" is named the same, so i was a bit confused there too. I leave this to your criteria.

Reply:
Yeah, it's something I think needs to be communicated better, but (I can't tell i posted this comment already in Github...) I have another pull request open that tries to make it a little more clear in the README (https://github.com/Islandora/islandora_solution_pack_video/pull/143). It's a struggle when we have to maintain that kind of legacy configuration.